runtime-sdk/src/context.rs

```diff
@@ -271,6 +275,31 @@ impl<'a, 'b> TxContext<'a, 'b> {
             .map(|x| x.downcast().expect("type should stay the same"))
             .unwrap_or_default()
     }
+
+    /// Fetches or sets a value associated with the transaction.
+    pub fn tx_value<V>(&mut self, key: &'static str) -> &mut V
+        where
+            V: Any + Default,
+    {
+        self.tx_values
+            .entry(key)
+            .or_insert_with(|| Box::new(V::default()))
+            .downcast_mut()
+            .expect("type should stay the same")
```

Comment:
I generally like to use `unwrap()` with a comment to prevent useless strings from filling up my `.text` section

Reply:
No strong preference, just make it consistent also in existing code.

Reply:
do we get good stack traces?

Reply:
this seems to be prevalent in places this PR won't :crossed_fingers: touch. let's split this out into a separate issue

Reply:
https://github.com/oasisprotocol/oasis-sdk/issues/99